Providers/CoinsProviderConfig.jsx

```diff
@@ -0,0 +1,123 @@
+// /* eslint-disable react/destructuring-assignment */
+import React, { Component, createContext } from "react";
+import { AsyncStorage, Alert } from "react-native";
+import axios from "react-native-axios";
+
+const { Provider, Consumer } = createContext();
+
+class CoinsProvider extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      coins: null,
+      date: null
+    };
+  }
+
+  async componentDidMount() {
+    const dateResult = await this.loadFromStorage();
+    try {
+      const resp = await axios.get("http://worldtimeapi.org/api/ip");
+
+      const date = new Date(resp.data.datetime);
+      const storedDate = new Date(dateResult);
+
+      if (this.state.date) {
+        if ((date - storedDate) / 1000 > 86400) {
+          AsyncStorage.setItem("date", JSON.stringify(date));
+          this.increaseCoins();
+        }
+      } else {
+        this.setState({ date: resp.data.datetime });
+        AsyncStorage.setItem("date", JSON.stringify(date));
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  loadFromStorage = async () => {
+    const storedCoins = await this.getStoredCoins();
+    const storedDate = await this.getStoredDate();
+    if (!storedCoins) {
+      AsyncStorage.setItem("coins", JSON.stringify(400));
+    }
+    this.setState({
+      coins: storedCoins === null ? 400 : storedCoins,
+      date: storedDate
+    });
+    console.warn(storedCoins);
+    return storedDate;
```

Comment:
dlaczego to ustawia dwie rzeczy ale zwraca tylko jedną, i nie wynika to z nazwy :P